Add contract compilation & collection docs (#2064)
foundry-rs/starknet-foundry: docs/src/testing/test-collection.md

In both of the above examples, you can
make the functions from `tests/common/utils.cairo` available in `tests/test_contract.cairo` 
by using a relative import: `use super::common::utils;`.

> ### How Contracts Are Collected

Reviewer: This should be in a separate file contract-collection.md or sth

When you call `snforge test`, one of the things that Forge does is that it calls Scarb, particularly `scarb build`.

> Scarb then builds all contracts you're using and stores them inside the `target/{current_profile}` directory

Reviewer: ```suggestion
It makes Scarb build all contracts from your package and save them to the `target/{current_profile}` directory
```

Reviewer 2: It's not strictly true since you can build external contracts as well

Reviewer: Yup, but then we can mention about it in the next section and it's all clear

(read more on [Scarb website](https://docs.swmansion.com/scarb/docs/extensions/starknet/contract-target.html)).

> Then, for each package, Foundry loads compiled contracts from that package and uses them to run tests.

Reviewer 2: It's confusing since you can't really tell what "package" are we referring to here

Author: Changed that to:
> Then, Foundry loads compiled contracts from each package (for all packages in the project) and uses them to run tests.

wdyt?

> ⚠️ **Warning**
> 

> > Make sure to define `[[target.starknet-contract]]` section in your Scarb.toml, otherwise Scarb won't build your contracts.

Reviewer: ```suggestion
> Make sure to define `[[target.starknet-contract]]` section in your `Scarb.toml`, otherwise Scarb won't build your contracts.
```
